src/smartstream/examples/map/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+use fluvio_smartstream::{smartstream, SimpleRecord};
+
+#[smartstream(map)]
+pub fn my_map(mut record: SimpleRecord) -> SimpleRecord {
```

Comment:
hmm.  not sure having mutation in the mapping

Reply:
This isn't a mutable reference, it is an owned SimpleRecord that we are binding as mutable. So we are given an owned value, we edit it, and then return it. We are not referencing something else and editing something behind the scenes.